packages/arb-token-bridge-ui/src/components/common/SidePanel.tsx

```diff
@@ -21,53 +21,89 @@ export const SidePanel = ({
   panelClassNameOverrides = '',
   scrollable = true
 }: SidePanelProps) => {
+  const [isClosing, setIsClosing] = useState(false)
+
+  function handleCloseStart() {
+    setIsClosing(true)
+  }
+
+  function handleCloseEnd() {
+    onClose?.()
+
+    // prevent flickering caused by race conditions
+    setTimeout(() => {
+      setIsClosing(false)
+    }, 0)
+  }
+
   return (
-    <Transition show={isOpen} duration="normal">
+    <Transition show={isOpen && !isClosing} as={Fragment}>
       <Dialog
         open={isOpen}
-        onClose={() => onClose?.()}
+        onClose={handleCloseStart}
```

Comment:
I'd wrap `handlecloseStart` with `useCallback` to avoid rerender

Reply:
Same for `handleCloseEnd`

Reply:
updated